domains/client/domain-operator/src/tests.rs

```diff
@@ -177,8 +177,24 @@ async fn test_domain_block_production() {
     assert_eq!(alice.client.info().best_hash, domain_block_hash);
 
     // Simply producing more block on fork C
-    ferdie.clear_tx_pool().await.unwrap();
-    produce_blocks!(ferdie, alice, 10).await.unwrap();
```

Comment:
Just a thought, I think it would be easier to update `produce_blocks` macro to handle this such that it will validate all the txns and fails by default.

Reply:
Do you mean calling `clear_tx_pool` inside `produce_blocks`?

Reply:
yeah, instead of clear just remove all the invalid txns. This is should be default for any test except those specific ones where we want to test the invalidity

Reply:
It may not help in this case, the tx in tx pool is supposed to be valid, the CI test failed due to an edge case that the `txpool-notifications` work has validated a tx in other stale forks and submitted the tx to the best fork and in the best fork the tx is invalid.

Calling `clear_tx_pool` before producing block does not help since there is still a rare condition that the worker can submit the invalid tx right after the call and before producing block, and the error should not happen to other tests since they don't construct multiple fork.